### PR TITLE
Test nullable body

### PR DESCRIPTION
**Pull request by pgcalixto**
Test the 2 possible variants for nullable body:
- type: ['object', 'null']
- type: 'object', nullable: true

See: https://github.com/fastify/fastify/issues/2618.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by pgcalixto** [comment]
One test job failed but I did not understand the error message:

```
> fastify@3.7.0 lint:typescript /Users/runner/work/fastify/fastify
> eslint -c types/.eslintrc.json types/**/*.d.ts test/types/**/*.test-d.ts

sh: line 1:  1313 Abort trap: 6           npm run lint:typescript
npm ERR! code ELIFECYCLE
npm ERR! errno 134
npm ERR! fastify@3.7.0 lint: `npm run lint:standard && npm run lint:typescript`
npm ERR! Exit status 134
npm ERR! 
npm ERR! Failed at the fastify@3.7.0 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

`sh: line 1:  1313 Abort trap: 6` 🤔

I tried to re-run the actions but it does not seem possible. 

Reference:
- https://github.com/fastify/fastify/pull/2648/checks?check_run_id=1304641150

**Review by mcollina — COMMENTED** [review]
Could you add some docs?

@jsumners wdyt?

**Comment by pgcalixto** [comment]
The docs already shows the nullable possibility:

- https://github.com/fastify/fastify/issues/2618#issuecomment-716127938
- https://www.fastify.io/docs/latest/Validation-and-Serialization/#validation

**Review by mcollina — APPROVED** [review]
lgtm

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Comment by jsumners** [comment]
Looking through the `light-my-request` source code, it look like `.inject` will have `payload: null` passed to the request handler. I do not think this simulates an empty body request as `curl -v -X POST server.intance/foo` would do.

I think we should have a test that uses a full fledged HTTP client to post an empty body to the server.

**Comment by pgcalixto** [comment]
Based on @jsumners's comment, I changed the tests to use `simple-get` instead of `fastify.inject`. I read the code in `simple-get` and it does not pass a null body unless explicitly written.

File: package.json [hunk 1]
```diff
@@ -114,8 +114,8 @@
   "devDependencies": {
     "@hapi/joi": "^17.1.1",
     "@sinonjs/fake-timers": "^6.0.1",
-    "@types/pino": "^6.0.1",
     "@types/node": "^14.10.3",
+    "@types/pino": "^6.0.1",
```
Comment (RafaelGSS) [inline package.json:118]: I would restore it

Comment (jsumners) [reply package.json:118]: Yeah, leave `package.json` out of this PR unless new dependencies need to be introduced/removed.

**Review by jsumners — APPROVED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (pgcalixto) [reply package.json]: Makes sense. Reverted it.

**Pull request merged**